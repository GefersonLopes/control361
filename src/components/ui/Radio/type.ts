export type RadioProps = {
  id: string;
  name: string;
  value: string;
  label: string;
  checked?: boolean;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
};
