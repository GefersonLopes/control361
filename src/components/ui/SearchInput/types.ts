import type { InputHTMLAttributes } from "react";

export interface SearchInputProps
  extends Omit<InputHTMLAttributes<HTMLInputElement>, "type" | "size"> {
  placeholder?: string;
  size?: "sm" | "md" | "lg";
}
