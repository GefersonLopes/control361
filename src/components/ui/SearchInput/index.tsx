import clsx from "clsx";
import { forwardRef } from "react";

import type { SearchInputProps } from "./types";

const SearchInput = forwardRef<HTMLInputElement, SearchInputProps>(
  ({ placeholder = "Buscarâ€¦", size = "md", className, ...props }, ref) => {
    const sizeClasses: Record<string, string> = {
      sm: "h-7 w-40 md:w-56",
      md: "h-9 w-56 md:w-72",
      lg: "h-11 w-72 md:w-96",
    };

    return (
      <input
        ref={ref}
        type="search"
        placeholder={placeholder}
        className={clsx(
          "rounded-md border border-[#8A939D] bg-transparent px-3 text-sm font-inter placeholder:text-[#8A939D] placeholder:font-light placeholder:text-xs placeholder:text-inter",
          "focus:border-cyan-600 focus:outline-none focus:ring-1 focus:ring-cyan-500",
          sizeClasses[size],
          className,
        )}
        {...props}
      />
    );
  },
);

SearchInput.displayName = "SearchInput";
export default SearchInput;
