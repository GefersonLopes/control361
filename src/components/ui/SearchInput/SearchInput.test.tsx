import "@testing-library/jest-dom";

import { fireEvent, render, screen } from "@testing-library/react";
import { createRef } from "react";

import SearchInput from "../SearchInput";
import type { SearchInputProps } from "./types";

describe("SearchInput component", () => {
  it("renders input with type search and default placeholder and forwards ref", () => {
    const ref = createRef<HTMLInputElement>();
    render(<SearchInput ref={ref} />);
    const input = screen.getByRole("searchbox");
    expect(input).toHaveAttribute("type", "search");
    expect(input).toHaveAttribute("placeholder", "Buscarâ€¦");
    expect(ref.current).toBe(input);
  });

  it("accepts custom placeholder", () => {
    render(<SearchInput placeholder="Pesquisar" />);
    const input = screen.getByPlaceholderText("Pesquisar");
    expect(input).toBeInTheDocument();
  });

  it("applies default md size classes", () => {
    render(<SearchInput />);
    const input = screen.getByRole("searchbox");
    expect(input).toHaveClass("h-9", "w-56", "md:w-72");
  });

  it.each<[SearchInputProps["size"], string[]]>([
    ["sm", ["h-7", "w-40", "md:w-56"]],
    ["md", ["h-9", "w-56", "md:w-72"]],
    ["lg", ["h-11", "w-72", "md:w-96"]],
  ])("applies correct classes for size '%s'", (size, expectedClasses) => {
    render(<SearchInput size={size} />);
    const input = screen.getByRole("searchbox");
    expectedClasses.forEach((cls) => {
      expect(input).toHaveClass(cls);
    });
  });

  it("merges additional className", () => {
    render(<SearchInput className="my-class" />);
    const input = screen.getByRole("searchbox");
    expect(input).toHaveClass("my-class");
  });

  it("forwards arbitrary props", () => {
    render(
      <SearchInput
        data-testid="search-test"
        aria-label="Pesquisar coisas"
        maxLength={10}
      />,
    );
    const input = screen.getByTestId("search-test");
    expect(input).toHaveAttribute("aria-label", "Pesquisar coisas");
    expect(input).toHaveAttribute("maxlength", "10");
  });

  it("calls onChange when input changes", () => {
    const handleChange = jest.fn();
    render(<SearchInput onChange={handleChange} />);
    const input = screen.getByRole("searchbox");
    fireEvent.change(input, { target: { value: "abc" } });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it("combines with snapshot", () => {
    const { asFragment } = render(
      <SearchInput
        placeholder="Snap"
        size="sm"
        className="snap-class"
        onChange={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
