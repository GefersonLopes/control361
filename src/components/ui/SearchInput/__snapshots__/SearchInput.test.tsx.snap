// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component combines with snapshot 1`] = `
<DocumentFragment>
  <input
    class="rounded-md border border-[#8A939D] bg-transparent px-3 text-sm font-inter placeholder:text-[#8A939D] placeholder:font-light placeholder:text-xs placeholder:text-inter focus:border-cyan-600 focus:outline-none focus:ring-1 focus:ring-cyan-500 h-7 w-40 md:w-56 snap-class"
    placeholder="Snap"
    type="search"
  />
</DocumentFragment>
`;
